_posts/2020-08-04-newpipe-turning-5.md

```diff
@@ -0,0 +1,31 @@
+
+I started NewPipe ... well five years ago. In summer 2015. Back then I had just finished school and had nothing to do until university started. As I was interested in alternative and open software already I joined a group of FSFE fellows that formed back then. 
+They initially showed me F-Droid which I thought was huge crap and unusable at first, however, after some time I wrapped my head around the idea of getting rid of Google and all the non-free software and just do whatever I find nice. So I replaced Google mail with mailbox.org. For my calendar/contacts/data foo I set up and an ownCloud instance on an ODROID c3. I also installed CyanogenMod OS without Google services on my phone I got back then and put F-Droid on it.
+Well, in the beginning not using Google on my phone was all kind of different. Finding the right apps for everything was hard, but eventually things worked out nice and I got used to my new setup.
+However, there was one thing I missed: Youtube.
+People told me to just use the Youtube mobile website, but have you ever tried it? It's horrible! It's ugly, slow and just weird. On the phones that existed back then this was really no fun, and so I soon looked for an alternative to it. I first tried to find some open source apps. There where none. Then I looked for apps that would just not require google play services, and also here I could not find anything. Luckily during my spare time I started to learn Android programming, and so I decided to write my own app. The idea was pretty simple: Take youtubedl and port it to Android.
+I didn't know how to port Python to Android. Therefore, I simply tried to figure out how Youtube-dl does its magic and rewrote the core logic in Java. That code would extract the video stream URL from a simple Youtube URL, and that's it. I wrapped an app around this code. One could use the android share function to enter a link into the app. It would then extract the stream URL and open the video in VLC player. Later I went on and wrote a simple scraper that would also search videos on the website and display the result. When you tap on the result the app would show you some details about the video, like the description and how much up/down votes it got. From here you could simply launch into VLC and watch the video.
+
+That was it. With this simple and slick app I didn't have to use the Youtube mobile website anymore and my problem was solved. I was done. I wanted to release my code on F-Droid to also let others have the opportunity to use it. So I named the app GnuTube and prepared everything for a release. I also added some features like the ability to download a video. Furthermore I set myself a goal of about 10 features/functions I wanted to properly implement before the initial release 1.0, however, as I showed the app to my FSFE friends they said they wouldn't want to wait and I should release it as soon as possible. Somehow, I thought,  'Well, v0.3 would pretty much represent my progress.'
```

Comment:
```suggestion
That was it. With this simple and slick app I didn't have to use the YouTube mobile website anymore and my problem was solved. I was done. I wanted to release my code on F-Droid to also let others have the opportunity to use it. So I named the app GnuTube and prepared everything for a release. I also added some features like the ability to download a video. Furthermore I set myself a goal of about 10 features/functions I wanted to properly implement before the initial release 1.0, however, as I showed the app to my FSFE friends they said they wouldn't want to wait and I should release it as soon as possible. Somehow, I thought,  'Well, v0.3 would pretty much represent my progress.'
```

@theScrabi What did you want to say in the third sentence? Did you release the code on GitHub or the app on F-Droid?

Reply:
I'll put both in it:
> I wanted to publish my code on GitHub and the app on F-Droid